lib/dynflow/action.rb

```diff
@@ -545,10 +548,13 @@ def execute_run(event)
           check_serializable :output
         end
 
+        # Only update the telemetry data if the step stopped for good
+        update_step_telemetry 'run' if [:success, :skipped].include? self.state
```

Comment:
This is called too early, as the metadata is recalculated here https://github.com/Dynflow/dynflow/blob/4c3e72b187037bb9e1607906518bb445ea774717/lib/dynflow/execution_plan/steps/abstract_flow_step.rb#L16, so it actually has values from previous run. 

Also, the execution time should be IMO updated every time the run is executed, but just with the time it took to execute it. The right place probabely should be in the `with_metadata_calculation` method.